src/middlewared/middlewared/api/v25_04_0/app.py

```diff
@@ -1,13 +1,376 @@
+    catalog_app: str | None = None
+    app_name: str = Field(pattern=r'^[a-z]([-a-z0-9]*[a-z0-9])?$', min_length=1, max_length=40)
+    '''
+    Application name must have the following:
+    1) Lowercase alphanumeric characters can be specified
+    2) Name must start with an alphabetic character and can end with alphanumeric character
+    3) Hyphen '-' is allowed but not as the first or last character
+    e.g abc123, abc, abcd-1232
+    '''
+    train: NonEmptyString = 'stable'
+    version: NonEmptyString = 'latest'
+
+
+class AppCreateResult(BaseModel):
+    result: AppEntry
+
+
+class AppUpdate(BaseModel):
+    values: Secret[dict] = Field(default_factory=dict)
+    custom_compose_config: Secret[dict] = Field(default_factory=dict)
+    custom_compose_config_string: Secret[LongString] = ''
+
+
+class AppUpdateArgs(BaseModel):
+    app_name: NonEmptyString
+    update: AppUpdate = AppUpdate()
+
+
+class AppUpdateResult(BaseModel):
+    result: AppEntry
+
+
+class AppDelete(BaseModel):
+    remove_images: bool = True
+    remove_ix_volumes: bool = False
+    force_remove_ix_volumes: bool = False
+    force_remove_custom_app: bool = False
+
+
+class AppDeleteArgs(BaseModel):
+    app_name: NonEmptyString
+    options: AppDelete = AppDelete()
+
+
+class AppDeleteResult(BaseModel):
+    result: Literal[True]
+
+
+class AppConfigArgs(BaseModel):
+    app_name: NonEmptyString
+
+
+class AppConfig(BaseModel):
+    model_config = ConfigDict(extra='allow')
+
+
+class AppConfigResult(BaseModel):
+    result: AppConfig
+
+
+class AppConvertToCustomArgs(BaseModel):
+    app_name: NonEmptyString
+
+
+class AppConvertToCustomResult(BaseModel):
+    result: AppEntry
+
+
+class AppStopArgs(BaseModel):
+    app_name: NonEmptyString
+
+
+class AppStopResult(BaseModel):
+    result: None
+
+
+class AppStartArgs(BaseModel):
+    app_name: NonEmptyString
+
+
+class AppStartResult(BaseModel):
+    result: None
+
+
+class AppRedeployArgs(BaseModel):
+    app_name: NonEmptyString
+
+
+class AppRedeployResult(BaseModel):
+    result: AppEntry
+
+
+class AppOutdatedDockerImagesArgs(BaseModel):
+    app_name: NonEmptyString
+
+
+class AppOutdatedDockerImagesResult(BaseModel):
+    result: list[NonEmptyString]
+
+
+class AppPullImages(BaseModel):
+    redeploy: bool = True
+
+
+class AppPullImagesArgs(BaseModel):
+    app_name: NonEmptyString
+    options: AppPullImages = AppPullImages()
+
+
+class AppPullImagesResult(BaseModel):
+    result: None
+
+
+class AppContainerIDOptions(BaseModel):
+    alive_only: bool = True
+
+
+class AppContainerIDArgs(BaseModel):
+    app_name: NonEmptyString
+    options: AppContainerIDOptions = AppContainerIDOptions()
+
+
+class ContainerDetails(BaseModel):
+    id: NonEmptyString
+    service_name: NonEmptyString
+    image: NonEmptyString
+    state: CONTAINER_STATES
+
+
+class AppContainerResponse(RootModel[dict[str, ContainerDetails]]):
+    pass
+
+
+class AppContainerIDResult(BaseModel):
```

Comment:
We should try and keep `Args` and `Result` model pairs together.

Reply:
Yes and that is indeed my intention but then for the result model we need to have another base model which is what is in between the args/result model. If we keep the args/result together then the models which are only used for these 2 will have to live some where else and it looked messy to me that way. So for example here is what i am proposing:

```
Args model

Some models required for result if any

Result model
```

Reply:
It does get hairy with lots of extra models so I can see where you're coming from. The NEP proposes this structure:
```
Entry model (if applicable)

All extra models

Args and Result pairs in alphabetical order
```
Granted, this does separate Args/Result models from their relevant helper models so I don't want to be the style police on this one.